components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIMWSDLReader.java

```diff
@@ -569,6 +585,47 @@ private void setServiceDefinition(Definition definition, API api) throws APIMana
 		}
 	}
 
+    /**
+     * Clear the actual service Endpoint and use Gateway Endpoint instead of the
+     * actual Endpoint for the given environment type.
+     *
+     * @param definition      {@link Definition} - WSDL4j wsdl definition
+     * @param api             API object
+     * @param environmentType gateway environment type
+     * @throws APIManagementException when error occurred getting WSDL address location
+     */
+    public void setServiceDefinition(Definition definition, API api, String environmentName, String environmentType)
+            throws APIManagementException {
+        Map serviceMap = definition.getAllServices();
+        Iterator serviceItr = serviceMap.entrySet().iterator();
+        URL addressURI;
+        try {
+            while (serviceItr.hasNext()) {
+                Map.Entry svcEntry = (Map.Entry) serviceItr.next();
+                Service svc = (Service) svcEntry.getValue();
+                Map portMap = svc.getPorts();
+                for (Object o : portMap.entrySet()) {
+                    Map.Entry portEntry = (Map.Entry) o;
+                    Port port = (Port) portEntry.getValue();
+
+                    List<ExtensibilityElement> extensibilityElementList = port.getExtensibilityElements();
+                    for (ExtensibilityElement extensibilityElement : extensibilityElementList) {
+                        addressURI = new URL(getAddressUrl(extensibilityElement));
+                        if (log.isDebugEnabled()) {
+                            log.debug("Address URI for the port:" + port.getName() + " is " + addressURI.toString());
+                        }
+                        String endpointTransport = determineURLTransport(addressURI.getProtocol(), api.getTransports());
+                        setAddressUrl(extensibilityElement, endpointTransport, api.getContext(), environmentName,
+                                environmentType);
+                    }
+                }
+            }
+        } catch (Exception e) {
+            String errorMsg = "Error occurred while getting the wsdl address location";
+            throw new APIManagementException(errorMsg, e);

```

Comment:
No need to define string variable to keep error message, just put in throw new APIManagementException("------", e);